jwst/outlier_detection/outlier_detection_step.py

```diff
@@ -220,11 +233,3 @@ def _get_asn_id(self, input_models):
                 asn_id=asn_id
             )
         return asn_id
-
-    def _set_status(self, input_models, status):
-        # this might be called with the input which might be a filename or path
```

Comment:
added support for it

Reply:
Thanks! I "unresolved" this as a reminder to myself to look at the string/filename support when it's pushed.

Reply:
sounds good, feel free to do so to other comments.  I resolve them as a way to check them off the list as I make the updates

Reply:
Was https://github.com/spacetelescope/jwst/pull/8683/commits/d47ef4b30b4d970e8c5da9a26a67bd0d9516e468#diff-1ee2b338c246e6e453c1e7f974b3b48e70e1c6b16a21ff14e6507c82fc9bf941R174 the commit to address this?

I don't see how `record_step_status` can handle filenames now that I look at it. With the changes in this PR when `record_step_status` is handed a filename:
- opens the file (using datamodels.open, so either a model or model container will be made)
- sets the step status on the opened model/container
- returns (but does not save the model/container)

Unless I'm missing something I suggest we remove the modifications to support filenames and re-introduce something like `_set_status` here. I'm open to other suggestions as well.

Reply:
I'm putting this back in the next iteration.  Let me know if you think it will work.  I just added a conditional to bypass datamodels.open if a ModelLibrary is passed in (although I don't think it would cause any issues to try calling datamodels.open on the library anyway)